tesseract/tesseract_planning/tesseract_motion_planners/test/ompl_planner_tests.cpp

```diff
@@ -304,6 +303,91 @@ TEST(OMPLMultiPlanner, OMPLMultiPlannerUnit)  // NOLINT
+    Eigen::Isometry3d pose;
+    fwd_kin_->calcFwdKin(pose, x);
+
+    Eigen::Vector3d z_axis = pose.matrix().col(2).template head<3>().normalized();
+
+    out[0] = z_axis.dot(normal_);
+  }
+
+private:
+  Eigen::Vector3d normal_;
+  tesseract_kinematics::ForwardKinematics::Ptr fwd_kin_;
+};
+
+TEST(OMPLConstraintPlanner, OMPLConstraintPlannerUnit)  // NOLINT
+{
+  EXPECT_EQ(ompl::RNG::getSeed(), SEED) << "Randomization seed does not match expected: " << ompl::RNG::getSeed()
+                                        << " vs. " << SEED;
+
+  // Step 1: Load scene and srdf
+  tesseract_scene_graph::ResourceLocator::Ptr locator =
+      std::make_shared<tesseract_scene_graph::SimpleResourceLocator>(locateResource);
+  Tesseract::Ptr tesseract = std::make_shared<Tesseract>();
+  boost::filesystem::path urdf_path(std::string(TESSERACT_SUPPORT_DIR) + "/urdf/lbr_iiwa_14_r820.urdf");
+  boost::filesystem::path srdf_path(std::string(TESSERACT_SUPPORT_DIR) + "/urdf/lbr_iiwa_14_r820.srdf");
+  EXPECT_TRUE(tesseract->init(urdf_path, srdf_path, locator));
+
+  // Step 2: Add box to environment
+  addBox(*(tesseract->getEnvironment()));
+
+  // Step 3: Create ompl planner config and populate it
+  auto kin = tesseract->getFwdKinematicsManagerConst()->getFwdKinematicSolver("manipulator");
+  std::vector<double> swp = start_state;
+  std::vector<double> ewp = end_state;
+
+  tesseract_motion_planners::OMPLMotionPlanner ompl_planner;
+
+  std::vector<tesseract_motion_planners::OMPLPlannerConfigurator::ConstPtr> planners;
+  planners.push_back(std::make_shared<tesseract_motion_planners::SBLConfigurator>());
+  planners.push_back(std::make_shared<tesseract_motion_planners::RRTConnectConfigurator>());
+
+  auto ompl_config =
+      std::make_shared<tesseract_motion_planners::OMPLPlannerFreespaceConfig>(tesseract, "manipulator", planners);
+
+  ompl_config->start_waypoint = std::make_shared<tesseract_motion_planners::JointWaypoint>(swp, kin->getJointNames());
+  ompl_config->end_waypoint = std::make_shared<tesseract_motion_planners::JointWaypoint>(ewp, kin->getJointNames());
+  ompl_config->collision_safety_margin = 0.02;
+  ompl_config->planning_time = 5.0;
+  ompl_config->max_solutions = 2;
+  ompl_config->longest_valid_segment_fraction = 0.01;
+
+  ompl_config->collision_continuous = true;
+  ompl_config->collision_check = true;
+  ompl_config->simplify = false;
+  ompl_config->n_output_states = 50;
+  ompl_config->constraint = std::make_shared<GlassUprightConstraint>(Eigen::Vector3d::UnitZ(), kin);
+
+  // Set the planner configuration
+  ompl_planner.setConfiguration(ompl_config);
+
+  tesseract_motion_planners::PlannerResponse ompl_planning_response;
+  tesseract_common::StatusCode status = ompl_planner.solve(ompl_planning_response);
+
+  if (!status)
+  {
+    CONSOLE_BRIDGE_logError("CI Error: %s", status.message().c_str());
+  }
+  EXPECT_TRUE(status);
+  EXPECT_TRUE(ompl_planning_response.joint_trajectory.trajectory.rows() >= ompl_config->n_output_states);
```

Comment:
It may also be worth looping through the trajectory, solving FK at each waypoint, and verifying the constraint was met

Reply:
Done.